### PR TITLE
Add top 4 datasource types

### PR DESCRIPTION
**Pull request by allensallinger**
Add top 4 datasource types

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -26,6 +26,13 @@ ruff = "==0.0.290"
 tomli = "^2.0.1"
 typing_extensions = ">=4.4.0"
 
+[tool.poetry.group.sql.dependencies]
+great-expectations = { version = "^0.17.15", extras = [
+    "postgresql",
+    "redshift",
+    "sqlalchemy",
+] }
```
Comment (Kilo59) [inline pyproject.toml:34]:
```suggestion
great-expectations = { version = "^0.17.15", extras = [
    "snowflake",
] }
```

Let's focus on snowflake first.
Try to lock/resolve the dependencies with just `snowflake`.

File: poetry.lock [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# This file is automatically @generated by Poetry 1.6.1 and should not be changed by hand.
```
Comment (Kilo59) [inline poetry.lock:1]: @allensallinger it looks like you are using an old version of poetry

Comment (allensallinger) [reply poetry.lock:1]: Upgraded the poetry version and I can now update the lock file.

File: pyproject.toml [hunk 3]
```diff
@@ -49,10 +50,7 @@ ignore_missing_imports = true
 follow_imports = 'normal'
 warn_redundant_casts = true
 show_error_codes = true
-enable_error_code = [
-    'ignore-without-code',
```
[the remark this answers is not in the document]
Comment (allensallinger) [reply pyproject.toml:53]: Yeah, I think my IDE auto formatted that. Let me see if I can get it to use two lines.

**Pull request merged**